R/test.R

```diff
@@ -43,99 +45,126 @@ isShinyTest <- function(text){
 #' @param filter If not `NULL`, only tests with file names matching this regular
 #'   expression will be executed. Matching is performed on the file name
 #'   including the extension.
+#' @param assert Logical value which determines if an error should be thrown if any error is captured.
+#' @param envir Parent testing environment in which to base the individual testing environments.
 #'
-#' @return A data frame classed with the supplemental class `"shinytestrun"`.
+#' @return A data frame classed with the supplemental class `"shiny_runtests"`.
 #'   The data frame has the following columns:
 #'
 #' | **Name** | **Type** | **Meaning** |
 #' | :-- | :-- | :-- |
 #' | `file` | `character(1)` | File name of the runner script in `tests/` that was sourced. |
 #' | `pass` | `logical(1)` | Whether or not the runner script signaled an error when sourced. |
-#' | `result` | any or `NA` | The return value of the runner, or `NA` if `pass == FALSE`. |
-#' | `error` | any or `NA` | The error signaled by the runner, or `NA` if `pass == TRUE`. |
+#' | `result` | any or `NA` | The return value of the runner |
 #'
 #' @details Historically, [shinytest](https://rstudio.github.io/shinytest/)
-#'   recommended placing tests at the top-level of the `tests/` directory. In
-#'   order to support that model, `testApp` first checks to see if the `.R`
-#'   files in the `tests/` directory are all shinytests; if so, just calls out
-#'   to [shinytest::testApp()].
+#'   recommended placing tests at the top-level of the `tests/` directory.
+#'   This older folder structure is not supported by runTests.
+#'   Please see [shinyAppTemplate()] for more details.
 #' @export
-runTests <- function(appDir=".", filter=NULL){
+runTests <- function(
+  appDir = ".",
+  filter = NULL,
+  assert = TRUE,
+  envir = globalenv()
+) {
+  # make sure shiny is available to use without `::`
```

Comment:
... "similar to runApp, so that shiny's functions are available in the UI and server code, and the test code."